### DISCUSSION TITLE
Create system-level or catalog of computed fields

### DISCUSSION THREAD

**Discussion by jedelman8** [nautobot/nautobot#1261] [Plugin (App) Ideas & Development]
### Overview 

Computed fields are great because they allow users to use Jinja filters to dynamically _compute_ data.  However, it may common to have computed fields be relevant across Nautobot installs that have similar designs or data use cases. This means that all of these users need to spend the time and research to create their respective computed fields and possibly Jinja filters that and their goal is the same.  The proposal is to explore system-level computed fields that can users can opt-in/out in `settings.py` or to build a catalog of computed fields that can be easily imported.   The main use case in mind comes from #146 in simplifying the creation of a computed field for anyone using Ansible although I'm sure there are many more use cases as well. 

### User Story

As Austin the the NAE,
I want to easily enable or opt-in or load computed fields that are relevant for my network design with the first one being Ansible connection type,
So that I can use consistent designs and patterns that align with the greater community and eliminate me from maintaining the needed logic that the community also has to deal with wrt computed fields.

**Comment by glennmatthews** [comment]
Just brainstorming here - it should be fairly straightforward to create a plugin that provides a "library" of computed field options and a config page (#935) for selecting which of these provided fields should be enabled/disabled.
